manifests/0000_30_cluster-api_09_admission-policies.yaml

```diff
@@ -152,18 +152,79 @@ data:
+data:
+  components: |-
+    apiVersion: admissionregistration.k8s.io/v1
+    kind: ValidatingAdmissionPolicyBinding
+    metadata:
+      name: "openshift-cluster-api-unsupported-aws-spec-fields"
+    spec:
+      policyName: "openshift-cluster-api-unsupported-aws-spec-fields"
+      validationActions: [Deny]
+      matchResources:
+        namespaceSelector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: In
+            values:
+            - openshift-cluster-api
+    ---
+    apiVersion: admissionregistration.k8s.io/v1
+    kind: ValidatingAdmissionPolicy
+    metadata:
+      name: "openshift-cluster-api-unsupported-aws-spec-fields"
+    spec:
+      failurePolicy: Fail
+      matchConstraints:
+        resourceRules:
+        - apiGroups:   ["infrastructure.cluster.x-k8s.io"]
+          apiVersions: ["v1beta2"]
+          operations:  ["CREATE", "UPDATE"]
+          resources:   ["awsmachines", "awsmachinetemplates"]
+      variables:
+        - name: machineSpec
+          expression: "object.kind == 'AWSMachine' ? object.spec : object.spec.template.spec"
+      validations:
+        - expression: "!has(variables.machineSpec.ami.eksLookupType)"
+          message: "spec.ami.eksLookupType is a forbidden field"
+        - expression: "!has(variables.machineSpec.imageLookupFormat)"
+          message: "spec.imageLookupFormat is a forbidden field"
+        - expression: "!has(variables.machineSpec.imageLookupOrg)"
+          message: "spec.imageLookupOrg is a forbidden field"
+        - expression: "!has(variables.machineSpec.imageLookupBaseOS)"
+          message: "spec.imageLookupBaseOS is a forbidden field"
+        - expression: "!has(variables.machineSpec.networkInterfaces)"
+          message: "spec.networkInterfaces is a forbidden field"
+        - expression: "!has(variables.machineSpec.uncompressedUserData)"
+          message: "spec.uncompressedUserData is a forbidden field"
+        - expression: "!has(variables.machineSpec.privateDnsName)"
+          message: "spec.privateDnsName is a forbidden field"
+        - expression: "!has(variables.machineSpec.ignition) || !has(variables.machineSpec.ignition.proxy)"
+          message: "spec.ignition.proxy is a forbidden field"
+        - expression: "!has(variables.machineSpec.ignition) || !has(variables.machineSpec.ignition.tls)"
+          message: "spec.ignition.tls is a forbidden field"
+        - expression: "!has(variables.machineSpec.securityGroupOverrides)"
+          message: "spec.securityGroupOverrides is a forbidden field"
+        - expression: >-
+            !has(variables.machineSpec.cloudInit) || 
+            (
+             !has(variables.machineSpec.cloudInit.secretCount) &&
+             !has(variables.machineSpec.cloudInit.secretPrefix) &&
+             !has(variables.machineSpec.cloudInit.insecureSkipSecretsManager) &&
+             !has(variables.machineSpec.cloudInit.secureSecretsBackend)
+            )
```

Comment:
I think just doing `!has(variables.machineSpec.cloudInit)` would suffice, we shouldn't need to check the fields within it

Reply:
This does not work because `cloudInit` is a non-pointer struct. The VAP always sees it as empty cloudInit{}.

One way to make it work is to add omitzero. This changes the API slightly. While for most clients this will unmarshal to the same struct as now, I am not confident this is safe to do. WDYT?

Reply:
I would raise an issue upstream and point this out, since there is a mutual exclusion between `cloudInit` and `ignition`, which can't be enforced correctly at the moment with this not being omitted correctly